src/recon_test/test_consistency_with_root.cxx

```diff
@@ -0,0 +1,547 @@
+ */
+
+#include "stir/ProjDataInfoCylindricalNoArcCorr.h"
+#include "stir/recon_buildblock/ProjMatrixByBin.h"
+#include "stir/recon_buildblock/ProjMatrixByBinUsingRayTracing.h"
+#include "stir/recon_buildblock/ProjMatrixElemsForOneBin.h"
+#include "stir/centre_of_gravity.h"
+#include "stir/listmode/CListModeDataROOT.h"
+#include "stir/listmode/CListRecord.h"
+#include "stir/IO/read_from_file.h"
+#include "stir/HighResWallClockTimer.h"
+#include "stir/DiscretisedDensity.h"
+#include "stir/VoxelsOnCartesianGrid.h"
+#include "stir/Succeeded.h"
+#include "stir/RunTests.h"
+#include "stir/Shape/GenerateImage.h"
+#include "boost/lexical_cast.hpp"
+
+#include "stir/warning.h"
+
+#include <stack>
+
+using std::cerr;
+using std::ifstream;
+
+START_NAMESPACE_STIR
+
+
+class WeightedCoordinate
+// This class is used to store the coordinates and weight
+{
+public:
+  WeightedCoordinate()
+      : coord(CartesianCoordinate3D<float>(0.f,0.f,0.f)), value(0.f)
+  {}
+  WeightedCoordinate(const CartesianCoordinate3D<float>& voxel_centre_v, const float value_v)
+      : coord(voxel_centre_v), value(value_v)
+  {}
+
+  //! Center of mass position in mm
+  CartesianCoordinate3D<float> coord;
+  //! Mass of all weights
+  float value;
+  //! COM standard deviation.
+};
+
+
+class ROOTConsistencyTests : public RunTests
+{
+public:
+  ROOTConsistencyTests()= default;
+
+  /*!
+   * \brief Run the tests with ROOT data
+
+   * \returns 0 if all tests passed, 1 otherwise
+   */
+  void run_tests() override;
+
+private:
+
+  void setup();
+  /*! Reads listmode event by event, computes the ProjMatrixElemsForOneBin (probabilities
+   * along a bin LOR). This method should include all tests/ setup for additional test such that the list data is
+   * read only once.
+   *
+   * Passes ProjMatrixElemsForOneBin (LOR) to test_nonTOF_LOR_closest_approach() and if fails,
+   * add 1 to `failed_events` (LOR's closest voxel was not within nonTOF_distance_threshold).
+   * Check if the number of `failed_events` is greater than half the number of tested events to pass the test.
+   */
+  void process_list_data();
+
+  void loop_through_list_events(shared_ptr<CListModeData> lm_data_sptr, const shared_ptr<ProjMatrixByBin>& proj_matrix_sptr,
+                                const CartesianCoordinate3D<float> original_coords);
+  /*! Given a ProjMatrixElemsForOneBin (probabilities), test if the closest voxel in the LOR to the original_coords
+   * is within nonTOF_distance_threshold distance. If it is, pass with true, otherwise fales.
+   * @param probabilities ProjMatrixElemsForOneBin object of a list mode event
+   * @param test_discretised_density_sptr Density containing a point source.
+   * @param original_coords Precomputed coordinates of the point source
+   * @param tolerance Precomputed voxel sizes - used in setting nonTOF_distance_threshold
+   * @return True if test passes, false if failed.
+   */
```

Comment:
```suggestion
  /*! Given a ProjMatrixElemsForOneBin (probabilities), test if the closest voxel in the LOR to the original_coords
   * is within nonTOF_distance_threshold distance. If it is, pass with true, otherwise fales.
   * @param probabilities ProjMatrixElemsForOneBin object of a list mode event
   * @return True if test passes, false if failed.
   */
```